tests.js

```diff
@@ -9,14 +9,51 @@
     return [template, values];
   }
 
+  function r(template, ...values) {
+    if (values.length !== 0) {
+      throw new Error();
+    }
+    return template.raw[0];
+  }
+
   const tests = [
+    // No interpolations
+    [...qpair`^foo\(bar\);\n$`, '',
+     r`/^foo\(bar\);\n$/`],
+    // No interpolations but flags
+    [...qpair`^foo\(bar\);\n$`, 'gi',
+     r`/^foo\(bar\);\n$/gi`],
+    // A single string into a block context.
     [...qpair`^${ 'foo' }$`, '',
-     '/^(?:foo)$/'],
+     r`/^(?:foo)$/`],
+    // Testing transitions between contexts.
     [...qpair`^([${ '\\' }${ /[a-z]/ }]{${ 42 }})${ /$/ }`, 'i',
-     '/^([\\\\a-z]{42})(?:$)/i'],
-    // See above for why this says "\$${" rather than "\${"
+     r`/^([\\a-z]{42})(?:$)/i`],
+    // Back-reference not scoped to containing RegExp
     [...qpair`^(#+)([^#\r\n]*)${ /\1/ }`, '',
      '/^(#+)([^#\\r\\n]*)(?:\\1)/'],
```

Comment:
\1 is not allowed in a template literal apparently.  Octal escape warning.  This may be a spec error.